src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -246,8 +251,15 @@
 
   }
 
+  public static Map<String, String> addDefaults(Map<String, String> props) {
+    ConcurrentMap<String, String> propsCopy = new ConcurrentHashMap<>(props);
+    propsCopy.putIfAbsent(STORAGE_CLASS_CONFIG, HdfsStorage.class.getName());
+    propsCopy.putIfAbsent(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG, AvroFormat.class.getName());
+    return propsCopy;
+  }
+
   public HdfsSinkConnectorConfig(Map<String, String> props) {
-    this(CONFIG_DEF, props);
+    this(CONFIG_DEF, addDefaults(props));
```

Comment:
Glad you mentioned. Reverting the commit that included the changes in the tests removed the test case I added for testing the defaults. 

Yes, it's needed because `getConfig` is static and it's only shadowed by that class. There's no real method override. So even if the base classes use `getConfig` to get the config def, which I don't think they do, the changes in `getConfig` only help for validation.